src/index.ts

```diff
@@ -284,6 +224,100 @@ function parseIntOption(option: string) {
   return parsed.toString();
 }
 
+function parseChainConfigOption(option: string): ChainConfigOptions {
+  // Split the option using ',' as the delimiter
+  const parts = option.split(",");
+
+  // Ensure there are at least two parts (rpc and deploymentBlock)
+  if (parts.length < 2) {
+    throw new InvalidArgumentError(
+      `Chain configuration must be in the format of <rpc>,<deploymentBlock>[,<watchdogTimeout>,<orderBookApi>], e.g. http://erigon.dappnode:8545,12345678,30,https://api.cow.fi/mainnet`
+    );
+  }
+
+  // Extract rpc and deploymentBlock from the parts
+  const rpc = parts[0];
+  // Ensure that the RPC is a valid URL
+  if (!isValidUrl(rpc)) {
+    throw new InvalidArgumentError(
+      `${rpc} must be a valid URL (RPC) (chainConfig)`
+    );
+  }
+
+  const rawDeploymentBlock = parts[1];
+  // Ensure that the deployment block is a number
+  const deploymentBlock = Number(rawDeploymentBlock);
+  if (isNaN(deploymentBlock)) {
+    throw new InvalidArgumentError(
+      `${rawDeploymentBlock} must be a number (deployment block)`
+    );
+  }
+
+  const rawWatchdogTimeout = parts[2];
+  // If there is a third part, it is the watchdogTimeout
+  const watchdogTimeout = parts.length > 2 ? Number(rawWatchdogTimeout) : 30;
+  // Ensure that the watchdogTimeout is a number
+  if (isNaN(watchdogTimeout)) {
+    throw new InvalidArgumentError(
+      `${rawWatchdogTimeout} must be a number (watchdogTimeout)`
+    );
+  }
```

Comment:
Also might make sense the number is positive

Reply:
Likewise, done.